tests/Doctrine/DBAL/Migrations/Tests/Tools/Console/Command/AbstractCommandTest.php

```diff
@@ -0,0 +1,166 @@
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $this->assertEquals($configuration, $this->invokeMigrationConfigurationGetter($input, $configuration));
+    }
+
+    /**
+     * Test if the migration configuration returns the connection from the helper set
+     */
+    public function testMigrationConfigurationReturnsConnectionFromHelperSet()
+    {
+        $input = $this->getMockBuilder('Symfony\Component\Console\Input\ArrayInput')
+            ->setConstructorArgs(array(array()))
+            ->setMethods(array('getOption'))
+            ->getMock();
+
+        $input->expects($this->any())
+            ->method('getOption')
+            ->with($this->logicalOr($this->equalTo('db-configuration'), $this->equalTo('configuration')))
+            ->will($this->returnValue(null));
+
+        $actualConfiguration = $this->invokeMigrationConfigurationGetter($input);
+
+        $this->assertInstanceOf('Doctrine\DBAL\Migrations\Configuration\Configuration', $actualConfiguration);
+        $this->assertEquals($this->getSqliteConnection(), $actualConfiguration->getConnection());
+    }
+
+    /**
+     * Test if the migration configuration returns the connection from the input option
+     */
+    public function testMigrationConfigurationReturnsConnectionFromInputOption()
+    {
+        $content = <<<EOF
+<?php
+
+return array('driver' => 'pdo_sqlite', 'memory' => true);
+EOF;
+        $filename = \tempnam(\sys_get_temp_dir(), 'migrations_test');
+        file_put_contents($filename, $content);
+
+        $input = $this->getMockBuilder('Symfony\Component\Console\Input\ArrayInput')
+            ->setConstructorArgs(array(array()))
+            ->setMethods(array('getOption'))
+            ->getMock();
+
+        $input->expects($this->any())
+            ->method('getOption')
+            ->with($this->logicalOr($this->equalTo('db-configuration'), $this->equalTo('configuration')))
+            ->will($this->returnCallback(function ($name) use ($filename) {
+                if ('db-configuration' === $name) {
+                    return $filename;
+                }
+                return null;
+            }));
+
+        $actualConfiguration = $this->invokeMigrationConfigurationGetter($input);
+
+        $this->assertInstanceOf('Doctrine\DBAL\Migrations\Configuration\Configuration', $actualConfiguration);
+        $this->assertEquals($this->getSqliteConnection(), $actualConfiguration->getConnection());
+
+        unlink($filename);
+    }
+
+    /**
+     * Test if the migration configuration returns values from the configuration file
+     */
+    public function testMigrationConfigurationReturnsConfigurationFileOption()
+    {
+        $content = <<<EOF
+name: "name"
+table_name: "migrations_table_name"
+migrations_namespace: "migrations_namespace"
+EOF;
+        $filename = \sys_get_temp_dir() . '/' . uniqid('migrations') . '.yml';
+        file_put_contents($filename, $content);
```

Comment:
It would be better to commit that file and reference it here... instead of dynamically create it in the test.